### PR TITLE
Retry transient errors when fetching cfssl*.

### PR DESCRIPTION
**Pull request by porridge**
**What this PR does / why we need it**:

Workaround for #55589.

**Release note**:
```release-note
NONE
```

/cc @shyamjvs FYI

### CODE REVIEW

File: hack/lib/util.sh [hunk 1]
```diff
@@ -737,12 +737,12 @@ function kube::util::ensure-cfssl {
     kernel=$(uname -s)
     case "${kernel}" in
       Linux)
-        curl -s -L -o cfssl https://pkg.cfssl.org/R1.2/cfssl_linux-amd64
-        curl -s -L -o cfssljson https://pkg.cfssl.org/R1.2/cfssljson_linux-amd64
+        curl --retry 10 -s -L -o cfssl https://pkg.cfssl.org/R1.2/cfssl_linux-amd64
```
Comment (shyamjvs) [inline hack/lib/util.sh]: Could you also add `--retry-delay`? To make sure that all attempts don't fail one after the other if server is down.

Comment (dims) [inline hack/lib/util.sh]: wanna add a `--retry-delay 5` as well?

Comment (dims) [reply hack/lib/util.sh]: LOL @shyamjvs great minds think alike :)

**Comment by porridge** [comment]
I'm not sure `--retry-delay` is a good idea. From the manpage:

>               When curl is about to retry a transfer, it will first  wait  one
>               second  and  then for all forthcoming retries it will double the
>               waiting time until it reaches 10 minutes which then will be  the
>               delay  between  the rest of the retries.  By using --retry-delay
>               you  disable  this  exponential  backoff  algorithm.  See   also
>               --retry-max-time  to  limit  the total time allowed for retries.
>               (Added in 7.12.3)

7.12.3 was relesed in 2004

**Comment by shyamjvs** [comment]
Interesting.. That's great!
/lgtm

@gmarek Could you approve the PR?

**Pull request merged**